wagtail/contrib/modeladmin/helpers/search.py

```diff
@@ -0,0 +1,49 @@
+import operator
+from functools import reduce
+
+from django.contrib.admin.utils import lookup_needs_distinct
+from django.db.models import Q
+
+from wagtail.search.backends import get_search_backend
+
+
+class ModelAdminSearchHandler:
+    def __init__(self, queryset, search_fields):
+        self.queryset = queryset
+        self.search_fields = search_fields
+
+    def search(self, search_term):
+        """
+        Returns a tuple containing a queryset to implement the search,
+        and a boolean indicating if the results may contain duplicates.
+        """
+        raise NotImplementedError()
+
```

Comment:
@seb-b I think we can simplify thing a little here by not even instantiating these classes. If you decorate `search()` with `@classmethod` and just allow it to accept `queryset`, `search_term` and `search_fields`... we can just use that directly in `get_search_results()`.

Reply:
The final class ended up a bit different than this, the `search_fields` is the only parameter and it's used by two methods